projects/observability/src/shared/components/bar-gauge/bar-gauge.component.ts

```diff
@@ -0,0 +1,102 @@
+import { ChangeDetectionStrategy, Component, ElementRef, Input, OnChanges, ViewChild } from '@angular/core';
+import { getPercentage, TypedSimpleChanges } from '@hypertrace/common';
+
+@Component({
+  selector: 'ht-bar-gauge',
+  styleUrls: ['./bar-gauge.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="bar-gauge">
+      <div *ngIf="this.title" class="title">{{ this.title | htDisplayTitle }}</div>
+      <div class="count">
+        {{ this.totalValue | number }} / {{ this.maxValue | number }}
```

Comment:
This will work. The idea behind the component is that its more like a stacked bar, where each segment is a part of a whole to have meaning. The total and max here are for all of the segments together.

Reply:
It has meaning, but visually if there are three segments and the label is treating them as one, what is the actual meaning of the individual segments?

Reply:
Here is the mock. I can only assume that we are going to subdivide and categorize whatever we are counting here. If we are looking at calls, maybe we can subdivide and categorize on types of calls. For the bar gauge where we show byte count, maybe we could subdivide on request vs response.

<img width="440" alt="Screen Shot 2020-12-21 at 9 31 43 AM" src="https://user-images.githubusercontent.com/45181984/102805226-c1fdd200-436f-11eb-99a7-582a581c2846.png">

Reply:
ok gotcha - it looks like we have both the total and the individual segment data - I was focussing more on the bottom part/legend. Is the plan for that to come in later?

Reply:
I'm having just so much fun iterating on this "simple" component... you know what, I'm going to go ahead and do the legend now too. Weeeeeee!